### PR TITLE
Code Execution API

### PR DESCRIPTION
**Pull request by samu80**
This API takes params (source_code, language_id) and resturns the output of the code

### CODE REVIEW

File: app/controllers/api/v1/executions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class ExecutionsController < ApiController
+      def submission_token
+        headers = { 'CONTENT_TYPE' => 'application/json', 'ACCEPT' => 'application/json' }
+        parameter = { source_code: params[:source_code], language_id: params[:language_id] }
+        response = JudgeZeroApi.new(parameter, headers).post('/submissions/?base64_encoded=false&wait=false')
+        token = JSON.parse(response.body)['token']
```
Comment (somnath77) [inline app/controllers/api/v1/executions_controller.rb:10]: you can write a support method to parse like we use in rspec

**Review by somnath77 — DISMISSED** [review]

**Review by varungadde99 — DISMISSED** [review]

File: app/controllers/api/v1/executions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class ExecutionsController < ApiController
+      def submission_token
+        headers = { 'CONTENT_TYPE' => 'application/json', 'ACCEPT' => 'application/json' }
```
Comment (rahulknojha) [inline app/controllers/api/v1/executions_controller.rb:7]: this is redundant, can't we add this in service itself ?

File: app/controllers/api/v1/executions_controller.rb (hunk 1 above)
Comment (rahulknojha) [inline app/controllers/api/v1/executions_controller.rb:10]: JSON.parse is also redundant, parse it in service and then return

**Review by rahulknojha — CHANGES_REQUESTED** [review]

**Review by rahulknojha — APPROVED** [review]

**Comment by samu80** [comment]
merged into development

**Pull request merged**